bigtable/tests/admin_integration_test.cc

```diff
@@ -12,185 +12,269 @@
-  auto table_list = admin.ListTables(admin_proto::Table::NAME_ONLY);
-  if (not table_list.empty()) {
-    throw std::runtime_error("Expected empty instance in integration test");
-  }
-  std::cout << "Initial ListTables() successful" << std::endl;
-}
+namespace bigtable {
+namespace testing {
+
+class AdminIntegrationTest : public TableIntegrationTest {
+ protected:
+  std::unique_ptr<bigtable::TableAdmin> table_admin_;
 
-void CheckCreateTable(bigtable::TableAdmin& admin, std::string table_id) {
-  auto table = admin.CreateTable(table_id, bigtable::TableConfig());
-  std::cout << "CreateTable(" << table_id << ") successful" << std::endl;
+  void SetUp() { CreateTableAdmin(); }
 
-  auto get_result = admin.GetTable(table_id);
-  if (table.name() != get_result.name()) {
-    std::ostringstream os;
-    os << "Mismatched names for GetTable(" << table_id << "): " << table.name()
-       << " != " << get_result.name();
-    throw std::runtime_error(os.str());
+  void TearDown() {}
+
+  void CreateTableAdmin() {
+    std::shared_ptr<bigtable::AdminClient> admin_client =
+        bigtable::CreateDefaultAdminClient(
+            bigtable::testing::TableTestEnvironment::project_id(),
+            bigtable::ClientOptions());
+    table_admin_ = absl::make_unique<bigtable::TableAdmin>(
+        admin_client, bigtable::testing::TableTestEnvironment::instance_id());
   }
-  std::cout << "GetTable(" << table_id << ") successful" << std::endl;
-}
 
-void CheckTableList(bigtable::TableAdmin& admin,
-                    std::vector<std::string> expected) {
-  auto table_list = admin.ListTables(admin_proto::Table::NAME_ONLY);
-  std::vector<std::string> actual;
-  for (auto const& tbl : table_list) {
-    actual.push_back(tbl.name());
+  bool TestForTableListCheck(vector<string> expectedTableList) {
+    vector<string> actualTableList;
+    vector<string> diffTableList;
+
+    auto tableList = table_admin_->ListTables(admin_proto::Table::NAME_ONLY);
+    for (auto const& tableName : tableList)
+      actualTableList.push_back(tableName.name());
+
+    // Sort actualTableList
+    sort(actualTableList.begin(), actualTableList.end());
+    // Sort expectedTableList
+    sort(expectedTableList.begin(), expectedTableList.end());
+
+    // Get the difference of expectedTableList and actualTableList
+    set_difference(expectedTableList.begin(), expectedTableList.end(),
+                   actualTableList.begin(), actualTableList.end(),
+                   back_inserter(diffTableList));
+
+    return diffTableList.empty();
   }
-  std::sort(actual.begin(), actual.end());
-  std::sort(expected.begin(), expected.end());
-  std::vector<std::string> differences;
-  std::set_difference(expected.begin(), expected.end(), actual.begin(),
-                      actual.end(), std::back_inserter(differences));
-  if (differences.empty()) {
-    std::cout << "ListTables() successful" << std::endl;
-    return;
+
+  bool checkTableSchema(admin_proto::Table const& actualTable,
```

Comment:
The naming convention is `CheckTableSchema` (note the initial uppercase), please fix.